src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -35,8 +34,8 @@ def initialize(server, port, options = {})
 
     @server = server
     @port = port
-    @use_ssl = options[:use_ssl]
-    @protocol = @use_ssl ? "https" : "http"
+    self.class.use_ssl = options[:use_ssl]
```

Comment:
@camlow325 If we do that, then the user will always have to specify a `hostcert`, a `hostprivkey`, and a `localcacert`, even if they're not making an https request. That's currently breaking our tests, which can easily be fixed, but will this break other projects creating instances of our client?